binary/src/Pos/Binary/Class/Core.hs

```diff
@@ -82,6 +112,26 @@ matchSize requestedSize lbl actualSize =
   when (actualSize /= requestedSize) $
     cborError (lbl <> " failed the size check. Expected " <> show requestedSize <> ", found " <> show actualSize)
 
+withSize :: Integral s => s -> a -> a -> a -> a -> a -> a
```

Comment:
Out loud thought -- as this seems to be used only in two places, would be possible/better to unroll the actual calls into a sort of static map, maybe without relying on CPP at all? I suspect the answer is no, but feel free to me as a rubber duck here 😉

Reply:
This function started life in @coot 's CBR-96 branch, where it gets considerably more use. So I think it should stay, or else whatever I do here will eventually become redundant if/when CBR-96 is merged.

Reply:
I found a bug in the second place where this was used, so I ended up making this static map anyway :)